File: Easy/Second Largest/second-largest.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>

using namespace std;

// } Driver Code Ends
//User function template for C++
class Solution{
public:	
	// Function returns the second
	// largest elements
	int print2largest(int arr[], int n) {
	    int sl = -1;
	    int l=0;
	    for(int i=1;i<n;i++){
	        if(arr[i]>arr[l]){
	            l=i;
	        }
	    }
	    
	    int largest = l;
	    for(int i=1;i<n;i++){
	        if(arr[i]>arr[largest]){
	            sl = largest;
	            largest=i;
	        }
	        if(arr[i]!=arr[largest]){
	            if(sl == -1 || arr[i]>arr[sl]){
	                sl=i;
	            }
	        }
	    }
	    if(sl == -1){
	        return -1;
	    }
	    else{
	        return arr[sl];
	    }
	    
	}
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        int arr[n];
        for (int i = 0; i < n; i++) {
            cin >> arr[i];
        }
        Solution ob;
        auto ans = ob.print2largest(arr, n);
        cout << ans << "\n";
    }
    return 0;
}

// } Driver Code Ends

File: Difficulty: Easy/Second Largest/second-largest.java
//{ Driver Code Starts
// Initial Template for Java
import java.io.*;
import java.util.*;
import java.util.stream.Collectors;

public class Main {

    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in);
        int t = Integer.parseInt(sc.nextLine());
        while (t-- > 0) {
            String[] arr1Str = sc.nextLine().split(" ");
            int[] arr = Arrays.stream(arr1Str).mapToInt(Integer::parseInt).toArray();
            Solution ob = new Solution();
            int ans = ob.getSecondLargest(arr);
            System.out.println(ans);

            System.out.println("~");
        }
    }
}

// } Driver Code Ends


// User function Template for Java

class Solution {
    //15-11-24
    public int getSecondLargest(int[] arr) {
        // Code Here
        int n = arr.length;
        int largest = arr[0];
        int slargest = -1;
        for(int i=1;i<n;i++){
            if(arr[i]>largest){
                slargest = largest;
                largest=arr[i];
            }
            if(arr[i]<largest && arr[i]>slargest){
                    slargest=arr[i];
            }
        }
        return slargest;
    }
}